pkg/acsengine/engine.go

```diff
@@ -294,7 +294,7 @@ func GenerateKubeConfig(properties *api.Properties, location string) (string, er
 	kubeconfig := string(b)
 	// variable replacement
 	kubeconfig = strings.Replace(kubeconfig, "{{WrapAsVerbatim \"variables('caCertificate')\"}}", base64.StdEncoding.EncodeToString([]byte(properties.CertificateProfile.CaCertificate)), -1)
-	if properties.OrchestratorProfile.KubernetesConfig.EnablePrivateCluster {
+	if *properties.OrchestratorProfile.KubernetesConfig.PrivateCluster.Enabled {
```

Comment:
We don't want the `*` reference here, if we want to have a simple boolean check (and we know that `properties.OrchestratorProfile.KubernetesConfig.PrivateCluster` is non-`nil`), we can evaluate `helpers.IsTrueBoolPointer(properties.OrchestratorProfile.KubernetesConfig.PrivateCluster.Enabled)` as the `if` condition.